### PR TITLE
NO-ISSUE: Starting tests for the update lifecycle

### PR DESCRIPTION
**Pull request by mangelajo**
This adds an additional E2E tests and some harness functions to help:
* Update devices
* Wait for devices status to be updated

### CODE REVIEW

File: test/e2e/agent/agent_updates_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+						if condition.Type == "Updating" && condition.Reason == "Rebooting" && condition.Status == "True" {
+							return true
+						}
+					}
+					return false
+				}, "2m")
+
+			harness.WaitForDeviceCondition(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference
```
Comment (mangelajo) [inline test/e2e/agent/agent_updates_test.go:76]: We should also check for:
```
 - lastTransitionTime: "2024-11-07T14:07:36.526489176Z"
  message: 'Updated to desired renderedVersion: 2'
  reason: Updated
  status: "False"
  type: Updating
  ```

here

File: test/e2e/agent/agent_updates_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+				}, "2m")
+
+			// TODO(hexfusion): we were expecting this update status not to be unknown at this point
+			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
```
Comment (hexfusion) [inline test/e2e/agent/agent_updates_test.go:80]: ack:  this is not yet implemented will track can you please ref the BZ
https://issues.redhat.com/browse/EDM-679

Comment (mangelajo) [reply test/e2e/agent/agent_updates_test.go:67]: done.

**Review by hexfusion — APPROVED** [review]
Looks good great start thanks!

**Comment by mangelajo** [comment]
@hexfusion updated a linter error, sorry.

**Review by hexfusion — APPROVED** [review]

**Pull request merged**